lms/djangoapps/courseware/module_render.py

```diff
@@ -846,8 +882,13 @@ def get_module_for_descriptor_internal(user, descriptor, student_data, course_id
     # that affects xblock visibility.
     user_needs_access_check = getattr(user, 'known', True) and not isinstance(user, SystemUser)
     if user_needs_access_check:
-        if not has_access(user, 'load', descriptor, course_id):
-            return None
+        access = has_access(user, 'load', descriptor, course_id)
+        if not access:
+            if access.user_message or access.user_fragment:
```

Comment:
can this be  
`if not access and not (access.user_message or access.user_fragment)`

Reply:
fixed